test/modules/node/esm-node.test.ts

```diff
@@ -88,16 +92,25 @@ console.log(typeof server.listen)
 
   const resolveStdio = await fsMock.exec('node ./resolve.cjs')
   expect(resolveStdio.stderr).toBe('')
+
   /**
    * @todo Take these expected export paths from package.json.
    * That should be the source of truth.
    */
+
+  /**
+   * @note Although the test requires the package in CJS,
+   * the "module-sync" condition allows loading the ESM build.
+   * This is supported in Node.js v20+.
+   */
   expect(resolveStdio.stdout).toMatch(
-    /^msw: (.+?)\/node_modules\/msw\/lib\/core\/index\.js/m,
+    /^msw: (.+?)\/node_modules\/msw\/lib\/core\/index\.mjs/m,
```

Comment:
Changing this test because using Node.js v20 as the default changes this behavior in how exports are resolved (it respects `module-sync` now).